tangelo/linq/tests/test_simulator.py

```diff
@@ -147,6 +147,43 @@ def test_get_exp_value_mixed_state(self):
             results[b] = sim.get_expectation_value(op1, circuit_mixed)
             np.testing.assert_almost_equal(results[b], reference, decimal=2)
 
+    def test_get_variance(self):
```

Comment:
I'm not sure I understand the point of testing `get_expectation_value` here.
You are not reusing those values anyway, as we recompute a brand new value in `get_variance` (exact, or from shots).

Maybe this was a little sanity check on your side to help with debugging and development ?
All I currently see is tests that do not test anything that was not tested already, unrelated to your feature, with each a probably to break depending on the samples.

Reply:
I think we should however test whatever that code is not using (e.g `_get_variance_from_frequencies` or `_get_variance_from_statevector`) explicitly to make sure both the underlying functions you have written are behaving as you'd expect.

Reply:
Since statevector is still based on ideal frequencies, we can handle all the functionality with `from_frequencies`.

Reply:
And those tests should be gone now :)